core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -189,9 +189,14 @@ Queue<HealthCheckContextGroup> contextGroupChain() {
         return contextGroupChain;
     }
 
-    private List<Endpoint> allHealthyEndpoints() {
-        final List<Endpoint> allHealthyEndpoints = new ArrayList<>();
+    @VisibleForTesting
+    List<Endpoint> allHealthyEndpoints() {
         synchronized (contextGroupChain) {
+            if (contextGroupChain.isEmpty()) {
+                return ImmutableList.of();
+            }
+
+            final List<Endpoint> allHealthyEndpoints = new ArrayList<>();
             final HealthCheckContextGroup newGroup = contextGroupChain.getLast();
```

Comment:
I thought using `pollLast` instead of `getLast` and returning empty on null is more natural, to only check contended `contextGroupChain` once (likely no difference in practice)

Reply:
Nice suggestion.